swim/member.go

```diff
@@ -128,6 +184,14 @@ func shouldProcessGossip(old *Member, gossip *Member) bool {
 		return false
 	}
 
+	// Only pick the gossip if the labels have a lower checksum then the checksum
+	// of the currently known labels
+	gossipLabelsChecksum := gossip.Labels.checksum()
+	oldLabelsChecksum := old.Labels.checksum()
+	if gossipLabelsChecksum < oldLabelsChecksum {
```

Comment:
Because the choice for the 'winning' view of a member then starts relying on the order in which gossips are received. eg. the last update to come in that is different will always win. As long as the member owning the keys is in the network it will be solved in the end because it will keep reasserting as long as 2 competing states are gossipped around, however if that member is not in the network anymore the network might have a hard time to converge as long as both gossips are maintained.

With this check it is order independent which state will be chosen by an actor in the network and therefore converge (I do not have formal proof for this, but in a [prior system](https://github.com/thanodnl/Evade) where I implemented an eventually consistent data structure I used the order independence to converge which worked well).